server/app/actions/ProgramDisabledAction.java

```diff
@@ -51,4 +65,30 @@ public CompletionStage<Result> call(Request req) {
 
     return delegate.call(req); // continute processing next step
   }
+
+  /** Get the program slug from flash key or from programId */
+  private Optional<String> getProgramSlug(Request request) {
+    Optional<String> programSlugOptional =
+        request.flash().get(FlashKey.REDIRECTED_FROM_PROGRAM_SLUG);
+
+    if (programSlugOptional.isPresent()) {
+      return programSlugOptional;
+    }
+
+    if (!settingsManifest.getFastforwardEnabled(request)
```

Comment:
I was concerned there would be some issues doing this all the time instead of just when FF is enabled. Otherwise we would probably have done this from the outset. Happy to always check for a programId if we know it'll behave correctly if FF is not on.